### PR TITLE
Event logs not emitted when dry run fails

### LINKED ISSUE #1762: Event logs not emitted when dry run fails
### Actual behavior

When dryRun fails, it does not emit event logs showing event stack trace.

A transaction with a fee lower than the expected minimum fee was ran against dryRun but a 500 error was returned.

**Request payload**
```
{
    "transaction": "0a05746f6b656e12087472616e73666572181620c0ac062a203972849f2ab66376a68671c10a00e8b8b67d880434cc65b04c6ed886dfa91c2c32270a0804000000000000001080c2d72f1a14fc18da54f6ce01bf31195548460361dfdb83c20522003a405ab9637b0b0316d3a78cbec14622677f4d3526af6ee224887bd6c0247e6a93ebe66f563111167dc33c8eaeda4bec3d2880c48b232dc2e8ef3749fc7293be1105",
    "skipVerify": false
}
```

**Response payload**
```
{
    "error": true,
    "message": "Server error: Invalid buffer length"
}

```

### Expected behavior

When dryRun fails, it should emit event logs showing event stack trace.

### Steps to reproduce

- Estimate transaction fees for a transaction against the `/transactions/estimate-fees` endpoint
  - Supply only the bare necessary inputs
- Use the estimations to update the transaction fees/params (when applicable), sign the transaction and submit it

### Which version(s) does this affect? (Environment, OS, etc...)

v0.7.0-beta.3

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1762 #1771

### How was it solved?
- [x] Remove `transaction.id` from estimate-fees and transaction-dryrun endpoints.'
- [x] Fix mismatch between transaction size and minFee, add missing `BYTE_FEE` in breakdown
- [x] Update docs and swagger
- [x] Update error handling for formatEvent
- [x] Add unit and integration test coverage for transactions estimate-fees endpoint   

### How was it tested?
- [x] Local

### CODE REVIEW

File: tests/integration/api_v3/http/transactionsEstimateFees.test.js [hunk 1]
```diff
@@ -55,6 +81,34 @@ describe('Post estimate-fees transactions API', () => {
 		expect(response).toMap(transactionEstimateFees);
 	});
 
+	describe.only('Test estimate-fees transactions for all transaction types', () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/transactionsEstimateFees.test.js:84]:
```suggestion
	describe('Test estimate-fees transactions for all transaction types', () => {
```

File: services/gateway/apis/http-version3/methods/transactionsDryRun.js [hunk 2]
```diff
@@ -30,7 +30,6 @@ module.exports = {
 				optional: false,
 				type: 'object',
 				props: {
-					id: { type: 'string' },
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/methods/transactionsDryRun.js:33]: We can leave it as is.

File: services/blockchain-connector/shared/sdk/formatter.js [hunk 3]
```diff
@@ -75,6 +73,18 @@ const formatTransaction = (transaction, additionalFee = 0) => {
...
 	);
 
+	// Calculate transaction size
+	const transactionBuffer = codec.encode(
+		txSchema,
+		{
+			...schemaCompliantTransaction,
+			fee: schemaCompliantTransaction.fee
+				? schemaCompliantTransaction.fee
+				: transactionMinFee,
```
Comment (nagdahimanshu) [inline services/blockchain-connector/shared/sdk/formatter.js:83]:
```suggestion
			fee: schemaCompliantTransaction.fee || transactionMinFee,
```

File: services/gateway/apis/http-version3/methods/transactionsDryRun.js (hunk 2 above)
Comment (priojeetpriyom) [reply services/gateway/apis/http-version3/methods/transactionsDryRun.js]: Applied.

File: tests/integration/api_v3/http/transactionsEstimateFees.test.js (hunk 1 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/transactionsEstimateFees.test.js]: Applied.

File: services/blockchain-connector/shared/sdk/formatter.js [hunk 4]
```diff
@@ -130,9 +140,12 @@ const formatEvent = (event, skipDecode) => {
 		eventData = event.data;
 	} else {
 		const eventDataSchema = getDataSchemaByEventName(event.name);
-		eventData = eventDataSchema
-			? codec.decodeJSON(eventDataSchema, Buffer.from(event.data, 'hex'))
-			: { data: event.data };
+		try {
+			eventData = eventDataSchema ? codec.decodeJSON(eventDataSchema, Buffer.from(event.data, 'hex')) : { data: event.data };
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/formatter.js]: Applied.

File: tests/integration/api_v3/http/transactionsEstimateFees.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/transactionsEstimateFees.test.js]: Applied.

File: services/blockchain-connector/shared/sdk/formatter.js (hunk 3 above)
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/formatter.js]: Applied.

File: docs/api/version3.md [hunk 5]
```diff
@@ -564,7 +563,8 @@ Request payload:
                     "additionalFees": { // optional - entries vary by command
                         "validatorRegistrationFee": "5000000", // only for pos:registerDelegate
                         "userAccountInitializationFee": "5000000", // only for token:transfer
-                        "escrowAccountInitializationFee": "5000000" // only for token:transferCrossChain
+                        "escrowAccountInitializationFee": "5000000", // only for token:transferCrossChain
+                        "bufferBytes": "6000"
```
Comment (sameersubudhi) [inline docs/api/version3.md:567]:
```suggestion
                        "bufferBytes": "6000" // temporary
```

File: docs/api/version3.md [hunk 6]
```diff
@@ -573,6 +573,7 @@ Request payload:
                     "ccmByteFee": "120000",
                     "additionalFees": {
                         "userAccountInitializationFee": "5000000",
+                        "bufferBytes": "6000"
```
Comment (sameersubudhi) [inline docs/api/version3.md:576]:
```suggestion
                        "bufferBytes": "6000" // temporary
```

**Review by nagdahimanshu — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍 Thank you for this contribution @priojeetpriyom and @nagdahimanshu. 🚀

**Pull request merged**